worker/caasapplicationprovisioner/application.go

```diff
@@ -196,16 +196,16 @@ func (a *appWorker) loop() error {
 			// Respond to state changes
 			err = handleChange()
 			if err != nil {
-				return nil
+				return errors.Trace(err)
```

Comment:
The prior behavior (mute errors) before your change looks quite dangerous... Should the changes in this method be backported to 2.9?

Reply:
yeah, I will backport these fixes to 2.9